hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/DBScanner.java

```diff
@@ -136,209 +154,168 @@ public static byte[] getValueObject(
+      sb.append(gson.toJson(o));
 
-  public static void setFileName(String name) {
-    DBScanner.fileName = name;
-  }
-
-  public static void setContainerId(long id) {
-    DBScanner.containerId = id;
-  }
+      iterator.get().next();
+      ++count;
+      if (withinLimit(count) && iterator.get().isValid()) {
+        // If this is not the last entry, append comma
+        sb.append(", ");
+      }
 
-  public static void setDnDBSchemaVersion(String version) {
-    DBScanner.dnDBSchemaVersion = version;
-  }
+      out.print(sb);
+    }
 
-  public static void setWithKey(boolean withKey) {
-    DBScanner.withKey = withKey;
+    if (withKey) {
+      // End JSON object
+      out.println(" }");
+    } else {
+      // End JSON array
+      out.println(" ]");
+    }
   }
 
-  public static void setShowCount(boolean showCount) {
-    DBScanner.showCount = showCount;
+  private boolean withinLimit(long i) {
+    return limit == -1L || i < limit;
   }
 
-  private static ColumnFamilyHandle getColumnFamilyHandle(
+  private ColumnFamilyHandle getColumnFamilyHandle(
             byte[] name, List<ColumnFamilyHandle> columnFamilyHandles) {
     return columnFamilyHandles
             .stream()
             .filter(
               handle -> {
                 try {
                   return Arrays.equals(handle.getName(), name);
-                    } catch (Exception ex) {
+                } catch (Exception ex) {
                   throw new RuntimeException(ex);
-                    }
+                }
               })
             .findAny()
             .orElse(null);
   }
 
-  private void constructColumnFamilyMap(DBDefinition dbDefinition) {
-    if (dbDefinition == null) {
-      System.out.println("Incorrect Db Path");
-      return;
-    }
-    this.columnFamilyMap = new HashMap<>();
-    DBColumnFamilyDefinition[] columnFamilyDefinitions = dbDefinition
-            .getColumnFamilies();
-    for (DBColumnFamilyDefinition definition:columnFamilyDefinitions) {
-      LOG.info("Added definition for table: {}", definition.getTableName());
-      this.columnFamilyMap.put(definition.getTableName(), definition);
-    }
-  }
-
-  @Override
-  public Void call() throws Exception {
-    List<ColumnFamilyDescriptor> cfs =
-        RocksDBUtils.getColumnFamilyDescriptors(parent.getDbPath());
-
-    final List<ColumnFamilyHandle> columnFamilyHandleList =
-        new ArrayList<>();
-    ManagedRocksDB rocksDB = ManagedRocksDB.openReadOnly(parent.getDbPath(),
-            cfs, columnFamilyHandleList);
-    this.printAppropriateTable(columnFamilyHandleList,
-           rocksDB, parent.getDbPath());
-    return null;
-  }
-
-  private void printAppropriateTable(
-          List<ColumnFamilyHandle> columnFamilyHandleList,
-          ManagedRocksDB rocksDB, String dbPath)
+  /**
+   * Main table printing logic.
+   * User-provided args are not in the arg list.
+   */
+  private void printTable(List<ColumnFamilyHandle> columnFamilyHandleList,
```

Comment:
I don't know what's correct format in Ozone for this type of scenarios but for me it should be either 
```
  private void printTable(List<ColumnFamilyHandle> columnFamilyHandleList,
                          ManagedRocksDB rocksDB, String dbPath)
      throws IOException, RocksDBException {
    ...
  }
```

or 
```
  private void printTable(
      List<ColumnFamilyHandle> columnFamilyHandleList,
      ManagedRocksDB rocksDB, String dbPath)
      throws IOException, RocksDBException 
  {
    ...
  }

  ```